src/nih_wayfinding/app/scripts/routing/directions-service.js

```diff
@@ -263,15 +249,41 @@
 
                 // add properties for audited edges
                 if (lastModified > 0) {
-                    if (step.hazards || step.crossSlope || (step.maxSlope > Config.warningMinimumGrade) ||
-                        (step.surface && (step.surface !== 'Concrete'))) {
-                        warnings.push('images/icons/icon-caution.svg');
+                    if (step.hazards) {
```

Comment:
These if blocks are getting long, might be about time to put them all in a separate private method.